csharp/src/Apache.Arrow/Arrays/ArrayData.cs

```diff
@@ -53,5 +54,24 @@ public sealed class ArrayData
             Buffers = buffers;
             Children = children;
         }
+
+        public void Dispose()
+        {
+            if (Buffers != null)
+            {
+                foreach (ArrowBuffer buffer in Buffers)
+                {
+                    buffer.Dispose();
+                }
+            }
+
+            if (Children != null)
+            {
+                foreach (ArrayData child in Children)
+                {
+                    child?.Dispose();
```

Comment:
I don't see any code that guards against having a `null` in the `Children` list, and it has a public constructor. So I assumed it can contain nulls. If we add a check, we can remove the `?.`.